tests/integration/devfile/utils/utils.go

```diff
@@ -186,3 +190,132 @@ func ExecWithRestartAttribute(projectDirPath, cmpName, namespace string) {
+
+	isDevfileTest := false
+
+	// if the odoV2Watch object is not empty, its a devfile test
+	if !reflect.DeepEqual(odoV2Watch, OdoV2Watch{}) {
+		isDevfileTest = true
+	}
+
+	startSimulationCh := make(chan bool)
+	go func() {
+		startMsg := <-startSimulationCh
+		if startMsg {
+			err := os.MkdirAll(filepath.Join(context, ".abc"), 0750)
+			Expect(err).To(BeNil())
+
+			err = os.MkdirAll(filepath.Join(context, "abcd"), 0750)
+			Expect(err).To(BeNil())
+
+			_, err = os.Create(filepath.Join(context, "a.txt"))
+			Expect(err).To(BeNil())
+
+			helper.DeleteDir(filepath.Join(context, "abcd"))
+
+			if isDevfileTest {
+				helper.ReplaceString(filepath.Join(context, "app", "app.js"), "Hello", "Hello odo")
+			} else {
+				if odoV1Watch.SrcType == "openjdk" {
+					helper.ReplaceString(filepath.Join(context, "src", "main", "java", "MessageProducer.java"), "Hello", "Hello odo")
+				} else {
+					helper.ReplaceString(filepath.Join(context, "server.js"), "Hello", "Hello odo")
+				}
+			}
+		}
+	}()
+
+	if !isDevfileTest {
+		flag = strings.TrimSpace(fmt.Sprintf("%s-app -v 4 %s", odoV1Watch.SrcType, flag))
+	}
+
+	success, err := helper.WatchNonRetCmdStdOut(
+		("odo watch " + flag + " --context " + context),
+		time.Duration(5)*time.Minute,
+		func(output string) bool {
+			if isDevfileTest {
+				stringsMatched := true
+
+				for _, stringToBeMatched := range odoV2Watch.StringsToBeMatched {
+					if !strings.Contains(output, stringToBeMatched) {
+						stringsMatched = false
+					}
+				}
+
+				if stringsMatched {
+					// Verify delete from component pod
+					getContainerExecListdir(odoV1Watch, odoV2Watch, runner, platform, project, isDevfileTest)
+					return true
+				}
+			} else {
+				curlURL := helper.CmdShouldPass("curl", odoV1Watch.RouteURL)
+				if strings.Contains(curlURL, "Hello odo") {
+					// Verify delete from component pod
+					getContainerExecListdir(odoV1Watch, odoV2Watch, runner, platform, project, isDevfileTest)
+				}
+				return strings.Contains(curlURL, "Hello odo")
+			}
+
+			return false
+		},
+		startSimulationCh,
+		func(output string) bool {
+			return strings.Contains(output, "Waiting for something to change")
+		})
+
+	Expect(success).To(Equal(true))
+	Expect(err).To(BeNil())
+
+	if !isDevfileTest {
+		// Verify memory limits to be same as configured
+		getMemoryLimit := runner.(helper.OcRunner).MaxMemory(odoV1Watch.SrcType+"-app", odoV1Watch.AppName, project)
+		Expect(getMemoryLimit).To(ContainSubstring("700Mi"))
+		getMemoryRequest := runner.(helper.OcRunner).MinMemory(odoV1Watch.SrcType+"-app", odoV1Watch.AppName, project)
+		Expect(getMemoryRequest).To(ContainSubstring("400Mi"))
+	}
+}
+
+func getContainerExecListdir(odoV1Watch OdoV1Watch, odoV2Watch OdoV2Watch, runner interface{}, platform, project string, isDevfileTest bool) string {
+	var stdOut string
+	sourcePath := "/projects/nodejs-web-app"
+
+	if platform == "kube" {
```

Comment:
Can you please use switch statement instead. Actually switch works much faster than equivalent if-else ladder. It is more readable. It is more manageable for having higher level of indentation than if. For example

```
switch platform {
	case "kube":
		// run the steps
		switch testType {
			case "devfile":
				// run the steps for devfile
			default:
				// run the steps for regular command test
		}
	case "docker":
		// run the steps
	default:
		fmt.Errorf("Platform %s is not supported", platform)
	}
```